File: clients/kotlin/src/main/kotlin/model/ColorFloat.kt
package model

import util.StreamUtil

class ColorFloat {
    var r: Float = 0.0f
    var g: Float = 0.0f
    var b: Float = 0.0f
    var a: Float = 0.0f
    constructor() {}
    constructor(r: Float, g: Float, b: Float, a: Float) {
        this.r = r
        this.g = g
        this.b = b
        this.a = a
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): ColorFloat {
            val result = ColorFloat()
            result.r = StreamUtil.readFloat(stream)
            result.g = StreamUtil.readFloat(stream)
            result.b = StreamUtil.readFloat(stream)
            result.a = StreamUtil.readFloat(stream)
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeFloat(stream, r)
        StreamUtil.writeFloat(stream, g)
        StreamUtil.writeFloat(stream, b)
        StreamUtil.writeFloat(stream, a)
    }
}

File: clients/javascript/model/color-float.js
class ColorFloat {
    constructor(r, g, b, a) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }
    static async readFrom(stream) {
        let r;
        r = await stream.readFloat();
        let g;
        g = await stream.readFloat();
        let b;
        b = await stream.readFloat();
        let a;
        a = await stream.readFloat();
        return new ColorFloat(r, g, b, a);
    }
    async writeTo(stream) {
        let r = this.r;
        await stream.writeFloat(r);
        let g = this.g;
        await stream.writeFloat(g);
        let b = this.b;
        await stream.writeFloat(b);
        let a = this.a;
        await stream.writeFloat(a);
    }
}
module.exports = { ColorFloat: ColorFloat }
